### PR TITLE
extended support for XmlSchemaComplexContentExtension in XSDToSchema.getStructField

### PR DESCRIPTION
**Pull request by miku872**
Extended support for XmlSchemaComplexContentExtension and changed the way extension elements were being added to baseStructField

### CODE REVIEW

**Comment by srowen** [comment]
How does this differ from #479 ?

**Comment by miku872** [comment]
Hey Srowen, 
I changed the way baseStructFields were being added to extension elements. Previously baseStructFields were added as a separate node. In this PR baseStructField elements are being added directly instead of being added as separate node. I hope that makes sense. Also I refactored some code.

File: src/main/scala/com/databricks/spark/xml/util/XSDToSchema.scala [hunk 1]
```diff
@@ -23,13 +23,15 @@ import org.apache.spark.annotation.Experimental
...
 import org.apache.ws.commons.schema.constants.Constants
+import scala.collection.mutable
 
 /**
  * Utility to generate a Spark schema from an XSD. Not all XSD schemas are simple tabular schemas,
  * so not all elements or XSDs are supported.
  */
 @Experimental
 object XSDToSchema {
+  val schemaNameToStructFieldMap = collection.mutable.Map[String, StructField]()
```
Comment (srowen) [inline src/main/scala/com/databricks/spark/xml/util/XSDToSchema.scala:34]: You can just write mutable.Map
This can't be a global mutable value though

**Comment by srowen** [comment]
Closed in favor of #483

**Pull request closed without merging**